File: scala/Convert.scala
// Convert.scala
// Simple string conversion to Base64.
//
// Author: Brian Nash
import java.nio.charset.StandardCharsets

object Convert {
  def main(args: Array[String]): Unit = {

    /** Determines if args are available
      *
      * @param args Run-time arguments
      * @return evalation of status of args being present
      */
    def hasArgs: Boolean = args.length == 1

    try {
      if (hasArgs) {
        val byteEncoded = args(0).trim.getBytes(StandardCharsets.UTF_8)
        val encoded = java.util.Base64.getEncoder.encodeToString(byteEncoded)
        println(s"Base64 encoded result: $encoded")
      } else {
        println("No input to encode")
      }
    } catch {
      case npe: NullPointerException =>
        println("Invalid string for encoding")
      case _: Throwable => println("Unknown issue converting string")
    }
  }
}

File: scala/Convert.js
/**
 * Encoding processor for hex strings to base64.
 * Run via `node Convert.js 45766964696e74`
 */


const converter = input => {
  const buffer = Buffer.from(input, 'hex');
  const base64 = buffer.toString('base64');
  console.log(`result: ${base64}`);
}

const hexToBase64 = () => {
  try {
    const args = process.argv.slice(2);
    if (args.length > 0) {
      converter(args[0])
    } else {
      console.error('No input for conversion');
    }
  } catch(error) {
    console.error(`Error caught during conversion: ${error}`)
  }
}

hexToBase64();